crates/sparrow-physical/src/plan.rs

```diff
@@ -0,0 +1,12 @@
+use crate::Step;
+
+/// A plan is a directed, acyclic graph of steps.
+///
+/// The plan is represented as an array of steps, with each step referencing
+/// it's children (inputs) by index. The array is topologically sorted so that
+/// all children have indices less than the step that references them.
+#[derive(Debug)]
```

Comment:
My thinking for now that (a) supporting `serde` is good for plans/steps/expressions since that lets us use JSON or YAML as a way to read/write plans locally and (b) we can create a protobuf that captures the structure of the plan and write a conversion.

I suspect the physical and logical plans will have basically the same structure, with some differences in terms of what kinds of expressions and steps they support. We could probably use the same protobuf with either a "superset enum" or some extension points. Will wait to see when we have the logical and are implementing the protobuf representation.